File: etamen-backend/app/Modules/Notifications/Infrastructure/Models/NotificationToken.php
<?php

namespace App\Modules\Notifications\Infrastructure\Models;

use App\Models\User;
use App\Modules\Notifications\Domain\Enums\NotificationDeviceType;
use App\Modules\Notifications\Domain\Enums\NotificationTokenProvider;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class NotificationToken extends Model
{
    protected $fillable = [
        'user_id',
        'token',
        'token_hash',
        'provider',
        'device_type',
        'device_name',
        'app_version',
        'locale',
        'timezone',
        'is_active',
        'last_seen_at',
        'metadata',
    ];

    protected function casts(): array
    {
        return [
            'provider' => NotificationTokenProvider::class,
            'device_type' => NotificationDeviceType::class,
            'is_active' => 'boolean',
            'last_seen_at' => 'datetime',
            'metadata' => 'array',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: etamen-flutter/lib/features/notifications/domain/entities/notification_token.dart
class NotificationToken {
  const NotificationToken({
    required this.id,
    required this.provider,
    required this.deviceType,
    this.deviceName,
    this.appVersion,
    this.locale,
    this.timezone,
    this.isActive = true,
    this.lastSeenAt,
    this.createdAt,
  });

  final int id;
  final NotificationTokenProvider provider;
  final NotificationDeviceType deviceType;
  final String? deviceName;
  final String? appVersion;
  final String? locale;
  final String? timezone;
  final bool isActive;
  final DateTime? lastSeenAt;
  final DateTime? createdAt;
}

enum NotificationTokenProvider {
  fcm('fcm'),
  apns('apns'),
  webPush('web_push'),
  local('local'),
  unknown('unknown');

  const NotificationTokenProvider(this.wireValue);

  final String wireValue;

  static NotificationTokenProvider fromWire(Object? value) {
    final normalized = value?.toString();
    return NotificationTokenProvider.values.firstWhere(
      (item) => item.wireValue == normalized,
      orElse: () => NotificationTokenProvider.unknown,
    );
  }
}

enum NotificationDeviceType {
  android('android'),
  ios('ios'),
  web('web'),
  unknown('unknown');

  const NotificationDeviceType(this.wireValue);

  final String wireValue;

  static NotificationDeviceType fromWire(Object? value) {
    final normalized = value?.toString();
    return NotificationDeviceType.values.firstWhere(
      (item) => item.wireValue == normalized,
      orElse: () => NotificationDeviceType.unknown,
    );
  }
}
